openhands/sdk/conversation/conversation.py

```diff
@@ -135,6 +148,10 @@ def run(self) -> None:
             if iteration >= self.max_iteration_per_run:
                 break
 
+            with self.state:
+                if self.state.agent_paused:
+                    break
+
```

Comment:
Doesn't seems necessary to me, since it will be break in the next loop anyway:
```suggestion
```

<img width="1966" height="1544" alt="Image" src="https://github.com/user-attachments/assets/cad5c468-a477-4975-b7d6-38f6cde3ad9c" />